File: 프로그래머스/lv1.체육복.py
# Level1: 체육복
def solution(n, lost, reserve):
    for student in range(1,n+1):
        if student in lost and student in reserve:
            lost.remove(student)
            reserve.remove(student)
    for student in range(1,n+1):
        if student in lost and student-1 in reserve:
            lost.remove(student)
            reserve.remove(student-1)
        elif student in lost and student+1 in reserve:
            lost.remove(student)
            reserve.remove(student+1)
    return n - len(lost)

File: 프로그래머스/lv1.체육복.js
// Level1: 체육복
function solution(n, lost, reserve) {
    let copy_lost = [...lost];
    lost = lost.filter(e => !reserve.includes(e));
    reserve = reserve.filter(e => !copy_lost.includes(e));
    let answer = n - lost.length
    lost.sort((a,b) => a-b);
    lost.map((e,i) => {
        if(reserve.includes(e-1)){
            const idx = reserve.indexOf(e-1);
            reserve.splice(idx,1);
            answer += 1;
        }
        else if(reserve.includes(e+1)) {
            const idx = reserve.indexOf(e+1);
            reserve.splice(idx,1);
            answer += 1;
        }
    })
    return answer;
}